repos/system_upgrade/common/actors/addupgradebootentry/libraries/addupgradebootentry.py

```diff
@@ -33,6 +33,12 @@ def add_boot_entry(configs=None):
             # otherwise the new boot entry will not be set as default
             # See https://bugzilla.redhat.com/show_bug.cgi?id=1764306
             run(['/usr/sbin/zipl'])
+
+        if debug:
+            # The kernelopts for target kernel are generated based on the cmdline used in the upgrade initramfs,
+            # therefore, if we enabled debug above, we need to explicitly remove it from the boot entry.
+            api.produce(TargetKernelCmdlineArgTasks(to_remove=[KernelCmdlineArg(key='debug')]))
```

Comment:
```suggestion
            # NOTE: the current behaviour drops the debug option even in case it is set on the original system
            # already. We do not check original 'persistent' kernel args and when speaking about the debug
            # option, it would be pretty rare and not so reasonable to have debug option enabled on production
            # systems so we are ok to drop it always when leapp is executed with --debug.
            api.produce(TargetKernelCmdlineArgTasks(to_remove=[KernelCmdlineArg(key='debug')]))
```